test/gtest/uct/test_pending.cc

```diff
@@ -0,0 +1,122 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2014.  ALL RIGHTS RESERVED.
+* Copyright (C) UT-Battelle, LLC. 2014. ALL RIGHTS RESERVED.
+* See file LICENSE for terms.
+*/
+
+extern "C" {
+#include <uct/api/uct.h>
+#include <ucs/time/time.h>
+}
+#include <ucs/gtest/test.h>
+#include "uct_test.h"
+
+class test_uct_pending : public uct_test {
+public:
+    void initialize() {
+        uct_test::init();
+
+        m_e1 = uct_test::create_entity(0);
+        m_e2 = uct_test::create_entity(0);
+
+        m_e1->connect(0, *m_e2, 0);
+        m_e2->connect(0, *m_e1, 0);
+
+        m_entities.push_back(m_e1);
+        m_entities.push_back(m_e2);
+    }
+
+    typedef struct pending_send_request {
+        uct_ep_h          ep;
+        uint64_t          hdr;
+        unsigned          buffer;     /* Send buffer */
+        size_t            length;     /* Total length, in bytes */
+        uct_pending_req_t uct;
+    } pending_send_request_t;
+
+    static ucs_status_t pending_am_handler(void *arg, void *data, size_t length,
+                                           void *desc) {
+
+        unsigned *counter = (unsigned *) arg;
+        uint64_t test_hdr = *(uint64_t *) data;
+        uint64_t actual_data = *(unsigned*)((char*)data + sizeof(test_hdr));
+
+        if ((test_hdr == 0xabcd) && (actual_data == (0xdeadbeef + *counter))) {
+            (*counter)++;
+        } else {
+            UCS_TEST_ABORT("Error in comparison in pending_am_handler. Counter: " << counter);
+        }
+
+        return UCS_OK;
+    }
+
+    static ucs_status_t pending_send_op(uct_pending_req_t *self) {
+
+        pending_send_request_t *req = ucs_container_of(self, pending_send_request_t, uct);
+        ucs_status_t status;
+
+        status = uct_ep_am_short(req->ep, 0, req->hdr, &req->buffer, req->length);
+        if (status == UCS_OK) {
+            free(req);
+        }
+        return status;
+    }
+
+    void cleanup() {
+        uct_test::cleanup();
```

Comment:
no need to do it, if cleanup is not defined parent class cleanup will be called automatically